src/addondirectory.h

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef ADDONDIRECTORY_H
+#define ADDONDIRECTORY_H
+
+#include <QObject>
+
+class QString;
+class QByteArray;
+class QDir;
+
+constexpr const char* ADDON_FOLDER = "addons";
+
+class AddonDirectory final : public QObject {
+  Q_OBJECT
+
+ public:
+  AddonDirectory();
+
+  bool getDirectory(QDir* dir);
```

Comment:
All of these methods can be `const`. Like: `bool getDirectory(QDir* dir) const;`